ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFSum.java

```diff
@@ -337,29 +345,35 @@ public GenericUDAFEvaluator getWindowingEvaluator(WindowFrameDef wFrameDef) {
         return null;
       }
 
-      return new GenericUDAFStreamingEvaluator.SumAvgEnhancer<HiveDecimalWritable, HiveDecimal>(
+      return new GenericUDAFStreamingEvaluator.SumAvgEnhancer<HiveDecimalWritable, Object[]>(
```

Comment:
Hi @scarlin-cloudera Why are we changing this to Object[] instead of HiveDecimal and following places?

Reply:
This matches very similar logic found in GenericUDAFAvg which also uses an Object[] 

The purpose is that we need to keep track of the sum while also keeping track whether the current window member is a null.  If all members in the window frame are null, we should treat the sum as null instead of 0, which is how the "preceding" rows behave.

Reply:
I probably should also add:  We can't just put a NULL in the window because of the way the algorithm works.

The sum algorithm keeps track of the running sum total within the window that is stored.  The sum of the items in the window is the new running sum minus the running total being booted out of the window. So we need an extra data structure to hold the NULL value within the window.